<!DOCTYPE html><html lang="zh-CN" data-default-color-scheme="auto"><head><meta charset="UTF-8"><link rel="apple-touch-icon" sizes="76x76" href="/images/favicon.png"><link rel="icon" href="/images/favicon.png"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=5,shrink-to-fit=no"><meta http-equiv="x-ua-compatible" content="ie=edge"><meta name="theme-color" content="#2f4154"><meta name="author" content="Aye10032"><meta name="keywords" content=""><meta name="description" content="主要记点参数脚本什么的"><meta property="og:type" content="article"><meta property="og:title" content="测序数据处理自用笔记"><meta property="og:url" content="https://www.aye10032.com/2023/08/11/2023-08-11-trim_goal/index.html"><meta property="og:site_name" content="鸽子窝"><meta property="og:description" content="主要记点参数脚本什么的"><meta property="og:locale" content="zh_CN"><meta property="og:image" content="https://www.aye10032.com/images/posts/trim_goal/wsl%E5%88%86%E6%94%AF.png"><meta property="og:image" content="https://www.aye10032.com/images/posts/trim_goal/conda.png"><meta property="og:image" content="https://www.aye10032.com/images/posts/trim_goal/fastC%E7%BB%93%E6%9E%9C.png"><meta property="og:image" content="https://www.aye10032.com/images/posts/trim_goal/multiqc%20result.png"><meta property="og:image" content="https://www.aye10032.com/images/posts/trim_goal/after%20trim.png"><meta property="og:image" content="https://www.aye10032.com/images/posts/trim_goal/%E7%B4%A2%E5%BC%95%E6%96%87%E4%BB%B6.png"><meta property="article:published_time" content="2023-08-11T10:00:05.000Z"><meta property="article:modified_time" content="2023-08-28T09:02:33.000Z"><meta property="article:author" content="Aye10032"><meta property="article:tag" content="学习"><meta property="article:tag" content="生物信息学"><meta name="twitter:card" content="summary_large_image"><meta name="twitter:image" content="https://www.aye10032.com/images/posts/trim_goal/wsl%E5%88%86%E6%94%AF.png"><meta name="twitter:creator" content="@Aye10032"><meta name="referrer" content="no-referrer-when-downgrade"><title>测序数据处理自用笔记 - 鸽子窝</title><link rel="stylesheet" href="https://lib.baomitu.com/twitter-bootstrap/4.6.1/css/bootstrap.min.css"><link rel="stylesheet" href="https://lib.baomitu.com/github-markdown-css/4.0.0/github-markdown.min.css"><link rel="stylesheet" href="https://lib.baomitu.com/hint.css/2.7.0/hint.min.css"><link rel="stylesheet" href="https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.css"><link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_hj8rtnfg7um.css"><link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_lbnruvf0jn.css"><link rel="stylesheet" href="/css/main.css"><link id="highlight-css" rel="stylesheet" href="/css/highlight.css"><link id="highlight-css-dark" rel="stylesheet" href="/css/highlight-dark.css"><script id="fluid-configs">var Fluid=window.Fluid||{};Fluid.ctx=Object.assign({},Fluid.ctx);var CONFIG={hostname:"www.aye10032.com",root:"/",version:"1.9.5",typing:{enable:!0,typeSpeed:70,cursorChar:"_",loop:!1,scope:[]},anchorjs:{enable:!0,element:"h1,h2,h3,h4,h5,h6",placement:"right",visible:"hover",icon:""},progressbar:{enable:!0,height_px:3,color:"#29d",options:{showSpinner:!1,trickleSpeed:100}},code_language:{enable:!0,default:"TEXT"},copy_btn:!0,image_caption:{enable:!0},image_zoom:{enable:!0,img_url_replace:["",""]},toc:{enable:!0,placement:"right",headingSelector:"h1,h2,h3,h4,h5,h6",collapseDepth:0},lazyload:{enable:!0,loading_img:"/images/loading.gif",onlypost:!1,offset_factor:2},web_analytics:{enable:!0,follow_dnt:!0,baidu:null,google:{measurement_id:"G-285089DP1E"},tencent:{sid:null,cid:null},woyaola:null,cnzz:null,leancloud:{app_id:null,app_key:null,server_url:null,path:"window.location.pathname",ignore_local:!1}},search_path:"/local-search.xml",include_content_in_search:!0};if(CONFIG.web_analytics.follow_dnt){var dntVal=navigator.doNotTrack||window.doNotTrack||navigator.msDoNotTrack;Fluid.ctx.dnt=dntVal&&(dntVal.startsWith("1")||dntVal.startsWith("yes")||dntVal.startsWith("on"))}</script><script src="/js/utils.js"></script><script src="/js/color-schema.js"></script><script async>Fluid.ctx.dnt||Fluid.utils.createScript("https://www.googletagmanager.com/gtag/js?id=G-285089DP1E",(function(){function a(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],a("js",new Date),a("config","G-285089DP1E")}))</script><meta name="generator" content="Hexo 6.3.0"></head><body><header><div class="header-inner" style="height:70vh"><nav id="navbar" class="navbar fixed-top navbar-expand-lg navbar-dark scrolling-navbar"><div class="container"><a class="navbar-brand" href="/"><strong>鸽子窝</strong> </a><button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation"><div class="animated-icon"><span></span><span></span><span></span></div></button><div class="collapse navbar-collapse" id="navbarSupportedContent"><ul class="navbar-nav ml-auto text-center"><li class="nav-item"><a class="nav-link" href="/"><i class="iconfont icon-home-fill"></i> <span>首页</span></a></li><li class="nav-item"><a class="nav-link" href="/archives/"><i class="iconfont icon-archive-fill"></i> <span>归档</span></a></li><li class="nav-item"><a class="nav-link" href="/categories/"><i class="iconfont icon-category-fill"></i> <span>分类</span></a></li><li class="nav-item"><a class="nav-link" href="/tags/"><i class="iconfont icon-tags-fill"></i> <span>标签</span></a></li><li class="nav-item"><a class="nav-link" href="/about/"><i class="iconfont icon-user-fill"></i> <span>关于</span></a></li><li class="nav-item"><a class="nav-link" href="/bangumis/index.html"><i class="iconfont icon-bilibili-fill"></i> <span>我的追番</span></a></li><li class="nav-item"><a class="nav-link" href="/links/"><i class="iconfont icon-link-fill"></i> <span>友链</span></a></li><li class="nav-item" id="search-btn"><a class="nav-link" target="_self" href="javascript:;" data-toggle="modal" data-target="#modalSearch" aria-label="Search"><i class="iconfont icon-search"></i></a></li><li class="nav-item" id="color-toggle-btn"><a class="nav-link" target="_self" href="javascript:;" aria-label="Color Toggle"><i class="iconfont icon-dark" id="color-toggle-icon"></i></a></li></ul></div></div></nav><div id="banner" class="banner" parallax="true" style="background:url(/images/background-cover.jpg) no-repeat center center;background-size:cover"><div class="full-bg-img"><div class="mask flex-center" style="background-color:rgba(0,0,0,.3)"><div class="banner-text text-center fade-in-up"><div class="h2"><span id="subtitle" data-typed-text="测序数据处理自用笔记"></span></div><div class="mt-3"><span class="post-meta mr-2"><i class="iconfont icon-author" aria-hidden="true"></i> Aye10032 </span><span class="post-meta"><i class="iconfont icon-date-fill" aria-hidden="true"></i> <time datetime="2023-08-11 10:00" pubdate>2023年8月11日 上午</time></span></div><div class="mt-1"><span class="post-meta mr-2"><i class="iconfont icon-chart"></i> 15k 字 </span><span class="post-meta mr-2"><i class="iconfont icon-clock-fill"></i> 125 分钟 </span><span id="busuanzi_container_page_pv" style="display:none"><i class="iconfont icon-eye" aria-hidden="true"></i> <span id="busuanzi_value_page_pv"></span> 次</span></div></div></div></div></div></div></header><main><div class="container-fluid nopadding-x"><div class="row nomargin-x"><div class="side-col d-none d-lg-block col-lg-2"><aside class="sidebar category-bar" style="margin-right:-1rem"><div class="category-list"><div class="category row nomargin-x"><a class="category-item list-group-item category-item-action col-10 col-md-11 col-xm-11" title="学习" id="heading-4ef520d6cd20ba4a727af08e17e4939e" role="tab" data-toggle="collapse" href="#collapse-4ef520d6cd20ba4a727af08e17e4939e" aria-expanded="true">学习 <span class="list-group-count">(12)</span> <i class="iconfont icon-arrowright"></i></a><div class="category-collapse collapse show" id="collapse-4ef520d6cd20ba4a727af08e17e4939e" role="tabpanel" aria-labelledby="heading-4ef520d6cd20ba4a727af08e17e4939e"><div class="category-post-list"></div><div class="category-sub row nomargin-x"><a class="category-subitem collapsed list-group-item category-item-action col-10 col-md-11 col-xm-11" title="unity" id="heading-439b3a25b555b3bc8667a09a036ae70c" role="tab" data-toggle="collapse" href="#collapse-439b3a25b555b3bc8667a09a036ae70c" aria-expanded="false">unity <span class="list-group-count">(1)</span> <i class="iconfont icon-arrowright"></i></a><div class="category-collapse collapse" id="collapse-439b3a25b555b3bc8667a09a036ae70c" role="tabpanel" aria-labelledby="heading-439b3a25b555b3bc8667a09a036ae70c"><div class="category-post-list"><a href="/2023/01/21/2023-01-21-Unity-Particle/" title="unity 粒子效果笔记" class="list-group-item list-group-item-action"><span class="category-post">unity 粒子效果笔记</span></a></div></div></div><div class="category-sub row nomargin-x"><a class="category-subitem collapsed list-group-item category-item-action col-10 col-md-11 col-xm-11" title="期末复习" id="heading-69f506b4426d281814d067770ad53047" role="tab" data-toggle="collapse" href="#collapse-69f506b4426d281814d067770ad53047" aria-expanded="false">期末复习 <span class="list-group-count">(7)</span> <i class="iconfont icon-arrowright"></i></a><div class="category-collapse collapse" id="collapse-69f506b4426d281814d067770ad53047" role="tabpanel" aria-labelledby="heading-69f506b4426d281814d067770ad53047"><div class="category-post-list"><a href="/2018/05/20/2018-05-20-PLC/" title="PLC期末题库" class="list-group-item list-group-item-action"><span class="category-post">PLC期末题库</span> </a><a href="/2018/05/18/2018-05-18-dawu/" title="大学物理上复习笔记" class="list-group-item list-group-item-action"><span class="category-post">大学物理上复习笔记</span> </a><a href="/2020/06/28/2020-06-28-IoT/" title="工业物联网复习" class="list-group-item list-group-item-action"><span class="category-post">工业物联网复习</span> </a><a href="/2018/05/05/2018-05-05-feC/" title="工程材料学复习笔记" class="list-group-item list-group-item-action"><span class="category-post">工程材料学复习笔记</span> </a><a href="/2020/06/13/2020-06-13-Gcode/" title="机床数控复习" class="list-group-item list-group-item-action"><span class="category-post">机床数控复习</span> </a><a href="/categories/%E5%AD%A6%E4%B9%A0/%E6%9C%9F%E6%9C%AB%E5%A4%8D%E4%B9%A0/" class="list-group-item list-group-item-action"><span class="category-post">More...</span></a></div></div></div><div class="category-sub row nomargin-x"><a class="category-subitem collapsed list-group-item category-item-action col-10 col-md-11 col-xm-11" title="机器学习" id="heading-61bb1751fd355596e307767d1927c855" role="tab" data-toggle="collapse" href="#collapse-61bb1751fd355596e307767d1927c855" aria-expanded="false">机器学习 <span class="list-group-count">(1)</span> <i class="iconfont icon-arrowright"></i></a><div class="category-collapse collapse" id="collapse-61bb1751fd355596e307767d1927c855" role="tabpanel" aria-labelledby="heading-61bb1751fd355596e307767d1927c855"><div class="category-post-list"><a href="/2020/12/29/2020-12-29-CNNmodles/" title="几种经典的CNN模型" class="list-group-item list-group-item-action"><span class="category-post">几种经典的CNN模型</span></a></div></div></div><div class="category-sub row nomargin-x"><a class="category-subitem list-group-item category-item-action col-10 col-md-11 col-xm-11" title="生物信息学" id="heading-9b17337cce03384cfde4fda8da0ff881" role="tab" data-toggle="collapse" href="#collapse-9b17337cce03384cfde4fda8da0ff881" aria-expanded="true">生物信息学 <span class="list-group-count">(3)</span> <i class="iconfont icon-arrowright"></i></a><div class="category-collapse collapse show" id="collapse-9b17337cce03384cfde4fda8da0ff881" role="tabpanel" aria-labelledby="heading-9b17337cce03384cfde4fda8da0ff881"><div class="category-post-list"><a href="/2023/08/28/2023-08-28-sam_format/" title="sam/bam文件格式信息" class="list-group-item list-group-item-action"><span class="category-post">sam/bam文件格式信息</span> </a><a href="/2023/08/11/2023-08-11-trim_goal/" title="测序数据处理自用笔记" class="list-group-item list-group-item-action active"><span class="category-post">测序数据处理自用笔记</span> </a><a href="/2023/05/24/2023-05-24-Bioinformatics/" title="生物信息学知识图谱" class="list-group-item list-group-item-action"><span class="category-post">生物信息学知识图谱</span></a></div></div></div></div></div></div></aside></div><div class="col-lg-8 nopadding-x-md"><div class="container nopadding-x-md" id="board-ctn"><div id="board"><article class="post-content mx-auto"><h1 id="seo-header">测序数据处理自用笔记</h1><p class="note note-info">本文最后更新于：2023年8月28日 上午</p><div class="markdown-body"><h1 id="转录组数据处理笔记">转录组数据处理笔记</h1><h2 id="一前置环境">一、前置环境</h2><h3 id="linux子系统">1.1 Linux子系统</h3><p>首先肯定是在Linux上进行，我这边稍微记录一下现在装Linux子系统的步骤<sup id="fnref:1" class="footnote-ref"><a href="#fn:1" rel="footnote"><span class="hint--top hint--rounded" aria-label="[https://learn.microsoft.com/en-us/windows/wsl/install](https://learn.microsoft.com/en-us/windows/wsl/install)">[1]</span></a></sup>：</p><p>以管理员模式运行Windows terminal，并执行以下指令：</p><figure class="highlight powershell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs powershell">wsl <span class="hljs-literal">--install</span><br></code></pre></td></tr></table></figure><p>安装完毕后，再次运行此指令，能够看到如下界面：</p><figure><img src="/images/posts/trim_goal/wsl分支.png" srcset="/images/loading.gif" lazyload alt="wsl分支"><figcaption aria-hidden="true">wsl分支</figcaption></figure><p>接下来根据提示，安装指定的分支，例如我选择Ubuntu22：</p><figure class="highlight powershell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs powershell">wsl <span class="hljs-literal">--install</span> <span class="hljs-literal">-d</span> Ubuntu<span class="hljs-literal">-22</span>.<span class="hljs-number">04</span><br></code></pre></td></tr></table></figure><p>运行期间会要求输入用户名密码等，跟着输入就行。</p><div class="note note-warning"><p>当使用完子系统之后，直接关闭窗口并不会结束子系统进程，这会导致其依然在系统内存中占用资源。</p><p>想要彻底关闭WSL，在power shell中执行以下指令：</p><figure class="highlight powershell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs powershell">wsl <span class="hljs-literal">--shutdonn</span><br></code></pre></td></tr></table></figure></div><h3 id="conda安装及镜像配置">1.2 conda安装及镜像配置</h3><p>首先安装miniconda<sup id="fnref:2" class="footnote-ref"><a href="#fn:2" rel="footnote"><span class="hint--top hint--rounded" aria-label="[https://conda.io/projects/conda/en/stable/user-guide/install/index.html](https://conda.io/projects/conda/en/stable/user-guide/install/index.html)">[2]</span></a></sup>，下载安装脚本，并赋予其相应权限，之后运行脚本即可：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs bash">wget https://repo.anaconda.com/miniconda/Miniconda3-py310_23.5.2-0-Linux-x86_64.sh<br>sudo <span class="hljs-built_in">chmod</span> 777 Miniconda3-py310_23.5.2-0-Linux-x86_64.sh<br>./Miniconda3-py310_23.5.2-0-Linux-x86_64.sh<br></code></pre></td></tr></table></figure><blockquote><p>具体文件名字视你下的版本而定</p></blockquote><p>脚本运行期间，基本上是一路yes到底就行了。</p><p>conda安装完毕之后，重启系统或者通过一下指令刷新环境变量：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">source</span> .bashrc<br></code></pre></td></tr></table></figure><p>接下来配置所使用的镜像<sup id="fnref:3" class="footnote-ref"><a href="#fn:3" rel="footnote"><span class="hint--top hint--rounded" aria-label="[https://mirrors.tuna.tsinghua.edu.cn/help/anaconda/](https://mirrors.tuna.tsinghua.edu.cn/help/anaconda/)">[3]</span></a></sup>，这里需要使用bioconda这个channel：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs bash">conda config --<span class="hljs-built_in">set</span> show_channel_urls <span class="hljs-built_in">yes</span><br>nano .condarc<br></code></pre></td></tr></table></figure><p>将配置文件修改如下：</p><figure class="highlight yaml"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><code class="hljs yaml"><span class="hljs-attr">channels:</span><br>  <span class="hljs-bullet">-</span> <span class="hljs-string">defaults</span><br><span class="hljs-attr">show_channel_urls:</span> <span class="hljs-literal">true</span><br><span class="hljs-attr">default_channels:</span><br>  <span class="hljs-bullet">-</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main</span><br>  <span class="hljs-bullet">-</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/r</span><br>  <span class="hljs-bullet">-</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/msys2</span><br>  <span class="hljs-bullet">-</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge</span><br>  <span class="hljs-bullet">-</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/bioconda</span><br><span class="hljs-attr">custom_channels:</span><br>  <span class="hljs-attr">conda-forge:</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud</span><br>  <span class="hljs-attr">msys2:</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud</span><br>  <span class="hljs-attr">bioconda:</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud</span><br>  <span class="hljs-attr">menpo:</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud</span><br>  <span class="hljs-attr">pytorch:</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud</span><br>  <span class="hljs-attr">pytorch-lts:</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud</span><br>  <span class="hljs-attr">simpleitk:</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud</span><br>  <span class="hljs-attr">deepmodeling:</span> <span class="hljs-string">https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/</span><br></code></pre></td></tr></table></figure><p>之后使用如下指令来清空缓存：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">conda clean -i<br></code></pre></td></tr></table></figure><p>现在可以创建一个新的环境了：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs bash">conda create --name bioinfo python=3.10<br>conda activate bioinfo<br></code></pre></td></tr></table></figure><p><img src="/images/posts/trim_goal/conda.png" srcset="/images/loading.gif" lazyload></p><h2 id="二质量分析">二、质量分析</h2><p>这里所使用到的软件为fastqc与multiqc，这俩其实用处差不多，只不过后者可以将多份数据进行整合。</p><h3 id="软件安装">2.1 软件安装</h3><p>直接通过conda安装即可：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">conda install fastqc multiqc<br></code></pre></td></tr></table></figure><h3 id="参数含义">2.2 参数含义</h3><h4 id="fastqc4">FastQC<sup id="fnref:4" class="footnote-ref"><a href="#fn:4" rel="footnote"><span class="hint--top hint--rounded" aria-label="[https://github.com/s-andrews/FastQC](https://github.com/s-andrews/FastQC)">[4]</span></a></sup></h4><ul><li><code>-h</code> 或 <code>--help</code> ：显示帮助信息</li><li><code>-o &lt;output_dir&gt;</code> 或 <code>--outdir &lt;output_dir&gt;</code>：指定输出目录，将生成的结果文件保存到指定的目录中</li><li><code>-f &lt;format&gt;</code> 或 <code>--format &lt;format&gt;</code>：指定输入文件的格式。常见的格式包括"fastq"、"bam"、"sam"等</li><li><code>-t &lt;threads&gt;</code> 或 <code>--threads &lt;threads&gt;</code>：指定线程数，用于加速分析。默认为1个线程</li><li><code>-c &lt;config_file&gt;</code> 或 <code>--config &lt;config_file&gt;</code>：指定配置文件，配置额外的参数等</li><li><code>-k &lt;kmersize&gt;</code> 或 <code>--kmersize &lt;kmersize&gt;</code>：指定用于检测存在的最大k-mer大小。默认为10</li><li><code>-adapters &lt;adapters_file&gt;</code>：指定adapter序列文件，用于adapter污染检测</li><li><code>-limits &lt;limits_file&gt;</code>：指定limits文件，用于制定可接受的范围，例如序列长度、GC含量等</li><li><code>-extract</code>：提取原始的FastQC zip文件，而不进行分析</li><li><code>-q</code> 或 <code>--quiet</code>：静默模式，只输出错误消息</li></ul><h4 id="multiqc5">MultiQC<sup id="fnref:5" class="footnote-ref"><a href="#fn:5" rel="footnote"><span class="hint--top hint--rounded" aria-label="[https://multiqc.info/](https://multiqc.info/)">[5]</span></a></sup></h4><ul><li><code>-h</code> 或 <code>--help</code> ：显示帮助信息。</li><li><code>-o &lt;output_dir&gt;</code> 或 <code>--outdir &lt;output_dir&gt;</code>：指定输出目录，将生成的报告文件保存到指定的目录中。</li><li><code>-c &lt;config_file&gt;</code> 或 <code>--config &lt;config_file&gt;</code>：指定配置文件，自定义报告生成的规则和设置。</li><li><code>-s</code> 或 <code>--fullnames</code>：显示完整的样本名称，而不仅仅是文件名部分。</li><li><code>--title &lt;report_title&gt;</code>：指定报告的标题。</li><li><code>-f</code> 或 <code>--force</code>：强制重新生成报告，覆盖已存在的报告文件。</li><li><code>--ignore &lt;regexp&gt;</code>：忽略与指定正则表达式匹配的文件。</li><li><code>--exclude &lt;regexp&gt;</code>：排除与指定正则表达式匹配的文件。</li><li><code>--no-ansi</code>：禁用ANSI转义序列，不使用彩色输出。</li><li><code>-q</code> 或 <code>--quiet</code>：静默模式，只输出错误消息。</li></ul><h3 id="质量检测">2.3 质量检测</h3><p>由于这两个软件都无法自动创建目录，因此需要手动创建输出目录：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">mkdir</span> result<br></code></pre></td></tr></table></figure><p>对于单个文件，使用fastqc进行分析，如：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">fastqc -o result -f fastq -t 8 raw_data/H1_1.fq.gz<br></code></pre></td></tr></table></figure><p>即可得到分析结果：</p><p><img src="/images/posts/trim_goal/fastC结果.png" srcset="/images/loading.gif" lazyload></p><p>批量处理一个文件夹下的数据：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">cd</span> raw_data<br><span class="hljs-built_in">ls</span> *.gz |xargs fastqc -o ../result/ -t 8<br></code></pre></td></tr></table></figure><p>同时，可以使用MultiQC对多个数据文件进行整合：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">multiqc -o result/ result/<br></code></pre></td></tr></table></figure><p><img src="/images/posts/trim_goal/multiqc%20result.png" srcset="/images/loading.gif" lazyload></p><h2 id="三数据清洗">三、数据清洗</h2><h3 id="软件安装-1">3.1 软件安装</h3><p>数据清洗使用的软件为Trim Galore和fastp，它前置依赖Cutadapt和FastQC</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">conda install cutadapt trim-galore fastp<br></code></pre></td></tr></table></figure><h3 id="参数含义-1">3.2 参数含义</h3><h4 id="trim-galore6">Trim Galore<sup id="fnref:6" class="footnote-ref"><a href="#fn:6" rel="footnote"><span class="hint--top hint--rounded" aria-label="[https://github.com/FelixKrueger/TrimGalore](https://github.com/FelixKrueger/TrimGalore)">[6]</span></a></sup></h4><ul><li><code>--quality</code>：指定去除低质量reads的阈值，默认为20。 作用：去除质量低于指定阈值的reads</li><li><code>--length</code>：指定去除短reads的长度阈值，默认为20。 作用：去除长度低于指定阈值的reads</li><li><code>--stringency</code>：指定adapter剪切的严格程度，默认为1。 作用：调整adapter剪切的严格程度，值越高表示更严格的剪切</li><li><code>--paired</code>：指定输入的测序数据是否为配对的，默认为单端测序。 作用：指定处理的测序数据是否为配对的，以决定是否进行配对数据处理</li><li><code>--phred33</code>：：选择phred33或者phred64，表示测序平台使用的Phred quality score</li><li><code>--fastqc</code>：生成每个输入文件的质量评估报告，同时内部调用fastqc软件。 作用：生成测序数据的质量评估报告，以便进行质量控制</li><li><code>--fastqc_args "&lt;ARGS&gt;"</code> fastqc的参数</li><li><code>--gzip</code>：指定输出文件是否进行压缩，默认为否。 作用：选择是否对输出文件进行压缩，可以减小文件大小</li><li><code>--cores</code>：指定使用的CPU核心数，默认为1。 作用：指定trim-galore使用的CPU核心数，用于加速处理速度</li><li><code>--output_dir</code>：输入目录。需要提前建立目录，否则运行会报错</li></ul><h4 id="fastp7">fastp<sup id="fnref:7" class="footnote-ref"><a href="#fn:7" rel="footnote"><span class="hint--top hint--rounded" aria-label="[https://github.com/OpenGene/fastp](https://github.com/OpenGene/fastp)">[7]</span></a></sup></h4><ul><li><code>-i, --in1</code>：输入read1</li><li><code>-o, --out1</code>：输出read1</li><li><code>-I, --in2</code>：输入read2</li><li><code>-O, --out2</code>：输出read2</li><li><code>-A, --disable_ adpter_ trimming</code>：不过滤接头序列,默认过滤</li><li><code>-f, --trim_ front1</code>：去掉read1头部的几bp序列 ,默认为0</li><li><code>-t, --trim _tail1</code>：去掉read1尾部的几bp序列,默认为0</li><li><code>-F, --trim_ front2</code>：去掉read2头部的几bp序列,默认为0</li><li><code>-T, --trim_ front2</code>：去掉read2尾部的几bp序列,默认为0</li><li><code>-q,--qualified_ quality_phred</code>：质量值低于q的碱基将被过滤,默认为15</li><li><code>-I, --length_ required</code>：长度低于l的序列将被过滤,默认为15</li><li><code>-n, --n_ base_ limit</code>：序列中含N碱基多余n个将被过滤,默认为5</li><li><code>-5,-3</code>：开启滑窗质量裁剪</li><li><code>-W</code>：滑动窗口大小，默认4</li><li><code>-M</code>：滑窗裁剪平均质量，默认20</li><li><code>-j, --json</code>：生成可读的json文件</li><li><code>-h, --html</code>：生成html质控报告</li></ul><h3 id="清洗">3.3 清洗</h3><h4 id="使用trim-galore">使用Trim Galore</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">mkdir</span> clean_data<br><span class="hljs-built_in">mkdir</span> clean_result<br></code></pre></td></tr></table></figure><p>对于单个文件：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">trim_galore --paired --cores 2 --phred33 --length 35 --stringency 1 --quality 25 --gzip --fastqc_args <span class="hljs-string">&quot;-o trim_result/ -f fastq -t 8&quot;</span> --output_dir after_trim/ raw_data/H1_1.fq.gz raw_data/H1_2.fq.gz<br></code></pre></td></tr></table></figure><p>批量处理：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">mkdir</span> clean_data<br><span class="hljs-built_in">mkdir</span> clean_result<br><br><span class="hljs-built_in">ls</span> raw_data/*_1.fq.gz &gt; 1<br><span class="hljs-built_in">ls</span> raw_data/*_2.fq.gz &gt; 2<br><br><span class="hljs-built_in">paste</span> 1 2 &gt;config<br><br><span class="hljs-built_in">cat</span> config | <span class="hljs-keyword">while</span> <span class="hljs-built_in">read</span> <span class="hljs-built_in">id</span><br>  <span class="hljs-keyword">do</span><br>    arr=(<span class="hljs-variable">$id</span>)<br>    fq1=<span class="hljs-variable">$&#123;arr[0]&#125;</span><br>    fq2=<span class="hljs-variable">$&#123;arr[1]&#125;</span><br>    trim_galore --paired --cores 2 --phred33 --length 35 --stringency 1 --quality 25 --gzip --fastqc_args <span class="hljs-string">&quot;-o clean_result/ -f fastq -t 8&quot;</span> --output_dir clean_data/  <span class="hljs-variable">$fq1</span> <span class="hljs-variable">$fq2</span><br>  <span class="hljs-keyword">done</span><br><br><span class="hljs-built_in">rm</span> 1<br><span class="hljs-built_in">rm</span> 2<br><span class="hljs-built_in">rm</span> config<br><br>multiqc -o clean_result/ clean_result/<br><br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;done&quot;</span><br></code></pre></td></tr></table></figure><p>批处理执行完之后即可得到清洗后的数据和报告：</p><p><img src="/images/posts/trim_goal/after%20trim.png" srcset="/images/loading.gif" lazyload></p><h4 id="使用fastp">使用fastp</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">rm</span> -rf clean_data clean_report<br><br><span class="hljs-built_in">mkdir</span> clean_data<br><span class="hljs-built_in">mkdir</span> clean_report<br><br><span class="hljs-built_in">cd</span> raw_data/<br><span class="hljs-built_in">ls</span> *_1.fq.gz &gt; 1<br><span class="hljs-built_in">ls</span> *_2.fq.gz &gt; 2<br><br><span class="hljs-built_in">paste</span> 1 2 &gt;config<br><br><span class="hljs-built_in">cat</span> config | <span class="hljs-keyword">while</span> <span class="hljs-built_in">read</span> <span class="hljs-built_in">id</span><br>  <span class="hljs-keyword">do</span><br>    arr=(<span class="hljs-variable">$id</span>)<br>    fq1=<span class="hljs-variable">$&#123;arr[0]&#125;</span><br>    fq2=<span class="hljs-variable">$&#123;arr[1]&#125;</span><br>    fq1_clean=$(<span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;<span class="hljs-variable">$fq1</span>&quot;</span> | sed <span class="hljs-string">&#x27;s/.fq.gz/_clean.fq.gz/&#x27;</span>)<br>    fq2_clean=$(<span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;<span class="hljs-variable">$fq2</span>&quot;</span> | sed <span class="hljs-string">&#x27;s/.fq.gz/_clean.fq.gz/&#x27;</span>)<br>    fastp -i <span class="hljs-variable">$fq1</span> -o ../clean_data/<span class="hljs-variable">$fq1_clean</span> -I <span class="hljs-variable">$fq2</span> -O ../clean_data/<span class="hljs-variable">$fq2_clean</span> -q 25 -l 35 --thread=8  -5 -3<br>  <span class="hljs-keyword">done</span><br><br><span class="hljs-built_in">rm</span> 1 2 config<br><br><span class="hljs-built_in">cd</span> ../clean_data<br><span class="hljs-built_in">ls</span> *_clean.fq.gz |xargs fastqc -o ../clean_report/ -t 8<br><span class="hljs-built_in">cd</span> ..<br>multiqc -o clean_report/ clean_report/<br><br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;done&quot;</span><br></code></pre></td></tr></table></figure><h2 id="四上游数据的比对计数">四、上游数据的比对计数</h2><h3 id="软件安装-2">4.1 软件安装</h3><h4 id="hisat2">hisat2</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">conda install hisat2 <br></code></pre></td></tr></table></figure><h4 id="samtools8">samtools<sup id="fnref:8" class="footnote-ref"><a href="#fn:8" rel="footnote"><span class="hint--top hint--rounded" aria-label="[https://github.com/samtools/samtools/blob/develop/INSTALL](https://github.com/samtools/samtools/blob/develop/INSTALL)">[8]</span></a></sup></h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><code class="hljs bash">sudo apt-get update<br>sudo apt-get install autoconf automake make gcc perl zlib1g-dev libbz2-dev liblzma-dev libcurl4-gnutls-dev libssl-dev libncurses5-dev<br><br>wget https://github.com/samtools/samtools/releases/download/1.18/samtools-1.18.tar.bz2<br>tar -xjvf samtools-1.18.tar.bz2<br><span class="hljs-built_in">cd</span> samtools-1.18<br>./configure<br>make<br>sudo make install<br></code></pre></td></tr></table></figure><h4 id="featurecounts">featureCounts</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><code class="hljs bash">wget https://zenlayer.dl.sourceforge.net/project/subread/subread-2.0.6/subread-2.0.6-Linux-x86_64.tar.gz<br>tar -zxvf subread-2.0.6-Linux-x86_64.tar.gz<br>nano ~/.bashrc<br></code></pre></td></tr></table></figure><p>在最下面添加一行：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">export</span> PATH=<span class="hljs-string">&quot;/home/aye/subread-2.0.6-Linux-x86_64/bin:<span class="hljs-variable">$PATH</span>&quot;</span><br></code></pre></td></tr></table></figure><p>使更改生效：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">source</span> ~/.bashrc<br></code></pre></td></tr></table></figure><h3 id="参数含义-2">4.2 参数含义</h3><h4 id="hisat2-1">hisat2</h4><ul><li><code>-x</code>：参考基因组索引文件的前缀。注意最后是/genome</li><li><code>-1</code>：双端测序结果的第一个文件。若有多组数据，使用逗号将文件分隔。Reads的长度可以不一致</li><li><code>-2</code>：双端测序结果的第二个文件。 若有多组数据，使用逗号将文件分隔，并且文件顺序要和-1参数对应。Reads的长度可以不一致</li><li><code>-S</code>：指定输出的SAM文件。</li><li><code>-t</code>：输出time</li><li><code>-p</code>：线程使用设置</li><li><code>-U</code>：单端数据文件。若有多组数据，使用逗号将文件分隔。可以和-1、 -2参数同时使用。 Reads的长度可以不一致</li><li><code>--sra-acc</code>：输入SRA登录号, 比如SRR353653, SRR353654。多组数据之间使用逗号分隔。HISAT将自动下载并识别数据类型，进行比对</li></ul><h4 id="samtools9">samtools<sup id="fnref:9" class="footnote-ref"><a href="#fn:9" rel="footnote"><span class="hint--top hint--rounded" aria-label="[samtools的安装和使用](https://www.jianshu.com/p/6b7a442d293f)">[9]</span></a></sup></h4><ul><li><code>view</code>：用于将sam文件转化为bam文件<ul><li><code>-b</code>：该参数设置输出 BAM 格式，默认下输出是 SAM 格式文件</li><li><code>-h</code>：默认下输出的 sam 格式文件不带 header，该参数设定输出sam文件时带 header 信息</li><li><code>-H</code>：仅仅输出文件的头文件</li><li><code>-S</code>：默认下输入是 BAM 文件，若是输入是 SAM 文件，则最好加该参数，否则有时候会报错。</li><li><code>-u</code>：该参数的使用需要有-b参数，能节约时间，但是需要更多磁盘空间。</li><li><code>-c</code>：仅输出匹配的统计记录</li><li><code>-L</code>：仅包括和bed文件存在overlap的reads</li><li><code>-o</code>：输出文件的名称</li><li><code>-F</code>：过滤flag，仅输出指定FLAG值的序列</li><li><code>-q</code>：比对的最低质量值，一般认为20就为unique比对了，可以结合上述-bF参数使用使用提取特定的比对结果</li><li><code>-@</code>：指使用的线程数</li></ul></li><li><code>sort</code>：对bam文件进行排序<ul><li><code>-n</code>：设定排序方式按short reads的ID排序。默认下是按序列在fasta文件中的顺序（即header）和序列从左往右的位点排序。</li><li><code>-m INT</code>：设置每个线程的最大内存，单位可以是K/M/G，默认是 768M。对于处理大数据时，如果内存够用，则设置大点的值，以节约时间。</li><li><code>-t TAG</code>：按照TAG值排序</li><li><code>-o FILE</code>：输出到文件中，加文件名</li></ul></li><li><code>index</code>：建立索引，并生成bai文件</li><li><code>flagstat</code>：输出比对统计结果<ul><li><code>-@</code>：指使用的线程数</li></ul></li></ul><h4 id="featurecounts10">featureCounts<sup id="fnref:10" class="footnote-ref"><a href="#fn:10" rel="footnote"><span class="hint--top hint--rounded" aria-label="[https://zh.wikibooks.org/wiki/生物信息学/featureCounts](https://zh.wikibooks.org/wiki/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/featureCounts)">[10]</span></a></sup></h4><ul><li><p><code>-a</code>：指定注释文件</p></li><li><p><code>-o</code>：指定输出文件</p></li><li><p><code>-p</code>：设置此参数后，使用fragments (或 templates)代替reads计数。只对paired-end reads有效</p></li><li><p><code>-t</code>：指定GTF注释文件中的feature类型。默认：<code>exon</code></p><ul><li><p>exon：外显子，基因组中包含的编码序列片段。</p></li><li><p>CDS (Coding Sequence)：编码序列，用于描述外显子中编码蛋白质的部分。</p></li><li><p>start_codon：起始密码子，表示编码蛋白质的起始位置。</p></li><li><p>stop_codon：终止密码子，表示编码蛋白质的终止位置。</p></li><li><p>UTR (Untranslated Region)：未翻译区域，编码序列中没有被翻译为蛋白质的区域。</p></li><li><p>transcript：转录本，表示基因的一个转录过程，包含一系列的外显子。</p></li><li><p>gene：基因，包含一个或多个转录本。</p></li><li><p>tRNA：转移RNA，一类用于将氨基酸传递到翻译中的蛋白质合成过程的分子。</p></li><li><p>rRNA：核糖体RNA，一类参与蛋白质合成的RNA分子。</p></li><li><p>snoRNA：小核仁RNA，参与核仁中的RNA加工和修饰。</p></li><li><p>miRNA：微RNA，一类通过调控基因表达参与调节细胞内RNA稳定性的小分子RNA。</p></li><li><p>snRNA：小核RNA，参与剪接反应和转录调控。</p></li></ul></li><li><p><code>-g</code>：指定GTF注释文件中属性的类型。默认：<code>gene_id</code></p><ul><li><p>gene_id：基因的唯一标识符。</p></li><li><p>transcript_id：转录本的唯一标识符。</p></li><li><p>gene_name：基因的名称。</p></li><li><p>transcript_name：转录本的名称。</p></li><li><p>exon_number：外显子的编号。</p></li><li><p>exon_id：外显子的唯一标识符。</p></li><li><p>protein_id：蛋白质序列的唯一标识符。</p></li><li><p>gene_biotype：基因的生物类型，例如编码蛋白质的基因、非编码RNA基因等。</p></li><li><p>transcript_biotype：转录本的生物类型，例如编码蛋白质的转录本、非编码RNA转录本等。</p></li><li><p>gene_source：基因注释来源，表示该基因的来源数据库或参考序列。</p></li><li><p>transcript_source：转录本注释来源，表示该转录本的来源数据库或参考序列。</p></li><li><p>gene_symbol：基因的符号名称。</p></li><li><p>transcript_symbol：转录本的符号名称。</p></li></ul></li><li><p><code>-f</code>：在feature水平计数 (如对外显子计数)</p></li><li><p><code>-O</code>：统计所有重叠的的meta-features的reads</p></li><li><p><code>-C</code>：对于比对到不同染色体或相同染色体的不同链的read对，不进行计数</p></li><li><p><code>-T</code>：使用线程</p></li></ul><h3 id="数据处理">4.3 数据处理</h3><h4 id="get-sam-file">get sam file</h4><p>首先使用hisat2比对基因组得到sam文件，这里需要先下载参考基因组，因为我这里用的是人类细胞，所以使用hg38：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">cd</span><br><span class="hljs-built_in">mkdir</span> -p reference/index/hisat<br><span class="hljs-built_in">cd</span> reference/index/hisat/<br>wget https://genome-idx.s3.amazonaws.com/hisat/hg38_genome.tar.gz<br>tar -zxvf hg38_genome.tar.gz<br></code></pre></td></tr></table></figure><blockquote><p>这玩意很大，下不下来也可以考虑使用别的方法下载后上传到服务器</p></blockquote><p>下载并解压后文件如下：</p><p><img src="/images/posts/trim_goal/索引文件.png" srcset="/images/loading.gif" lazyload></p><p>接下来比对基因组</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">hisat2 --dta -t -x ../../reference/index/hisat/hg38/genome -1 H1_1_clean.fq.gz -2 H1_2_clean.fq.gz -p 4 -S hisat2_hg38data/H1.sam<br></code></pre></td></tr></table></figure><blockquote><p>这里genome是参考基因组文件的前缀</p></blockquote><h4 id="sam-to-bam">sam to bam</h4><p>之后，使用samtools进行进一步操作，首先是将sam文件转化为bam文件：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">samtools view -@ 8 -bS -o hisat2_hg38data/H1_1.bam hisat2_hg38data/H1_1.sam<br></code></pre></td></tr></table></figure><h4 id="sort-and-index">sort and index</h4><p>接下来对bam进行排序:</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">samtools <span class="hljs-built_in">sort</span> -@ 8 -o hisat2_hg38data/H1_1.sort.bam hisat2_hg38data/H1_1.bam<br></code></pre></td></tr></table></figure><p>之后对使用排序好的文件建立索引</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">samtools index hisat2_hg38data/H1_1.sort.bam<br></code></pre></td></tr></table></figure><p>最后，将比对结果输入到文件中保存：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">samtools flagstat -@ 8 -O json hisat2_hg38data/H1_1.sort.bam &gt; hisat2_hg38data/H1_1.sort.flagsata<br></code></pre></td></tr></table></figure><h4 id="count-table">count table</h4><p>使用featureCounts获得counts.txt：</p><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><code class="hljs bash">featureCounts -a ../reference/gtf/Homo_sapiens.GRCh38.110.chr.gtf.gz -g gene_id -t gene -O -p -T 8 -C -o counts/counts.txt hisat2_hg38data/*.sort.bam<br></code></pre></td></tr></table></figure><h4 id="批量处理">批量处理：</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br></pre></td><td class="code"><pre><code class="hljs bash"><span class="hljs-built_in">rm</span> -rf hisat2_hg38data/<br><span class="hljs-built_in">mkdir</span> hisat2_hg38data<br><br><span class="hljs-built_in">cd</span> clean_data/<br><span class="hljs-built_in">ls</span> *1_clean.fq.gz &gt; 1<br><span class="hljs-built_in">ls</span> *2_clean.fq.gz &gt; 2<br><span class="hljs-built_in">paste</span> 1 2 &gt;config<br><br><span class="hljs-built_in">cat</span> config | <span class="hljs-keyword">while</span> <span class="hljs-built_in">read</span> <span class="hljs-built_in">id</span><br>  <span class="hljs-keyword">do</span><br>    arr=(<span class="hljs-variable">$id</span>)<br>    fq1=<span class="hljs-variable">$&#123;arr[0]&#125;</span><br>    fq2=<span class="hljs-variable">$&#123;arr[1]&#125;</span><br>    output=$(<span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;<span class="hljs-variable">$fq1</span>&quot;</span> | sed <span class="hljs-string">&#x27;s/_1_clean.fq.gz/.sam/&#x27;</span>)<br>    <span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;hisat2 --dta -t -x ../../reference/index/hisat/hg38/genome -1 <span class="hljs-variable">$fq1</span> -2 <span class="hljs-variable">$fq2</span> -p 8 -S ../hisat2_hg38data/<span class="hljs-variable">$output</span>&quot;</span><br>    hisat2 --dta -t -x ../../reference/index/hisat/hg38/genome -1 <span class="hljs-variable">$fq1</span> -2 <span class="hljs-variable">$fq2</span> -p 8 -S ../hisat2_hg38data/<span class="hljs-variable">$output</span><br>  <span class="hljs-keyword">done</span><br>  <br><span class="hljs-built_in">rm</span> 1 2 config<br><br><span class="hljs-built_in">cd</span> ../hisat2_hg38data<br><br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;start sam to bam and sorting...&quot;</span><br><span class="hljs-built_in">ls</span> *.sam | xargs -I &#123;&#125; sh -c <span class="hljs-string">&#x27;samtools view -@ 8 -bS &#123;&#125; | samtools sort -@ 8 -o $(basename &#123;&#125; .sam).sort.bam&#x27;</span><br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;done, removing sam file...&quot;</span><br><span class="hljs-built_in">ls</span> *.sam | xargs <span class="hljs-built_in">rm</span><br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;done&quot;</span><br>tree<br><br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;start indexing...&quot;</span><br><span class="hljs-built_in">ls</span> *.sort.bam | xargs samtools index -M <br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;done&quot;</span><br><br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;saving to flagsata...&quot;</span><br><span class="hljs-built_in">ls</span> *.sort.bam | xargs -I &#123;&#125; sh -c <span class="hljs-string">&#x27;samtools flagstat -@ 8 &#123;&#125; &gt; $(basename &#123;&#125; .bam).flagstat&#x27;</span><br><span class="hljs-built_in">echo</span> <span class="hljs-string">&quot;done&quot;</span><br><br>multiqc ./<br><br><span class="hljs-built_in">cd</span> ..<br><span class="hljs-built_in">rm</span> -rf counts<br><span class="hljs-built_in">mkdir</span> counts<br><br>featureCounts -a ../reference/gtf/Homo_sapiens.GRCh38.110.chr.gtf.gz -g gene_id -t gene -O -p -T 8 -C -o counts/counts.txt hisat2_hg38data/*.sort.bam<br></code></pre></td></tr></table></figure><h3 id="结果解读11">4.4 结果解读<sup id="fnref:11" class="footnote-ref"><a href="#fn:11" rel="footnote"><span class="hint--top hint--rounded" aria-label="[基因组质量评估:(五)mapping法：2.samtools计算mapping rate](https://zhuanlan.zhihu.com/p/545367874)">[11]</span></a></sup></h3><h4 id="hisat2-2">hisat2</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><code class="hljs bash">19429766 reads; of these: <span class="hljs-comment"># reads总数</span><br>  19429766 (100.00%) were paired; of these: <span class="hljs-comment"># 配对的reads数量</span><br>    1066192 (5.49%) aligned concordantly 0 <span class="hljs-built_in">times</span> <span class="hljs-comment"># 一致地比对了0次的reads数量</span><br>    14853850 (76.45%) aligned concordantly exactly 1 time <span class="hljs-comment"># 一致地比对了1次的reads数量</span><br>    3509724 (18.06%) aligned concordantly &gt;1 <span class="hljs-built_in">times</span> <span class="hljs-comment"># 一致地比对了大于1次的reads数量</span><br>    ----<br>    1066192 pairs aligned concordantly 0 <span class="hljs-built_in">times</span>; of these: <span class="hljs-comment"># 一致地比对了0次的reads数量中：</span><br>      54954 (5.15%) aligned discordantly 1 time <span class="hljs-comment"># 不一致地比对了1次的reads数量</span><br>    ----<br>    1011238 pairs aligned 0 <span class="hljs-built_in">times</span> concordantly or discordantly; of these: <span class="hljs-comment">#一致或不一致地比对了0次的reads数量中：</span><br>      2022476 mates make up the pairs; of these: <span class="hljs-comment"># 配对的reads数量中：</span><br>        1211647 (59.91%) aligned 0 <span class="hljs-built_in">times</span> <span class="hljs-comment">#比对0次的数量</span><br>        607196 (30.02%) aligned exactly 1 time <span class="hljs-comment">#比对1次的数量</span><br>        203633 (10.07%) aligned &gt;1 <span class="hljs-built_in">times</span> <span class="hljs-comment">#比对大于1次的数量</span><br>96.88% overall alignment rate <span class="hljs-comment"># mapping rate，由mapped reads number/total reads number的比例计算得到</span><br>[bam_sort_core] merging from 20 files and 4 in-memory blocks...<br></code></pre></td></tr></table></figure><h4 id="samtools">samtools</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><code class="hljs bash">51231959 + 0 <span class="hljs-keyword">in</span> total (QC-passed reads + QC-failed reads) <span class="hljs-comment">#共有51231959条reads通过QC+0条reads未通过QC，后面的信息行中+后的都是代表QC没通过的reads的数量。</span><br>12372427 + 0 secondary<br>0 + 0 supplementary<br>0 + 0 duplicates<br>50020312 + 0 mapped (97.63% : N/A) <span class="hljs-comment"># 97.63%比例的reads mapping到参考序列上，这就是mapping record rate</span><br>38859532 + 0 paired <span class="hljs-keyword">in</span> sequencing<br>19429766 + 0 read1 <span class="hljs-comment"># 双端reads中read1的总数</span><br>19429766 + 0 read2 <span class="hljs-comment"># 双端reads中read2的总数</span><br>36727148 + 0 properly paired (94.51% : N/A) <span class="hljs-comment"># 94.51%比例的reads成对的映射上</span><br>37160066 + 0 with itself and mate mapped <span class="hljs-comment"># read映射上但配对read没映射上的数量</span><br>487819 + 0 singletons (1.26% : N/A) <span class="hljs-comment"># 1.26%比例的read没映射上的同时，配对read映射上了</span><br>289948 + 0 with mate mapped to a different chr <span class="hljs-comment"># reads和配对reads映射到不同染色体的情况下的reads数量</span><br>205767 + 0 with mate mapped to a different chr (mapQ&gt;=5) <span class="hljs-comment"># reads和配对reads映射到不同染色体，且映射质量大于等于5的情况下的reads数量</span><br></code></pre></td></tr></table></figure><h2 id="五表达水平计算">五、表达水平计算</h2><p>经由上一步，得到了counts.txt，接下来就基于它进行进一步的计算。</p><p>虽然主流都是R语言，不过我比较属性python，反正都是大数据处理，没差别（</p><h3 id="count的结构">5.1 count的结构</h3><p>我这里根据上面的参数，生成的count结构如下所示：</p><ul><li>Geneid：基因的唯一标识符</li><li>Chr：基因所在的染色体</li><li>Start：基因在染色体上的起始位置</li><li>End：基因在染色体上的终止位置</li><li>Strand：基因在染色体上的定向，正义链为"+", 反义链为"-"</li><li>Length：基因的长度，单位为碱基</li><li>H1、H2：基因在样品中的表达量</li></ul><h3 id="rpkmfpkm和tpm12">5.2 RPKM、FPKM和TPM<sup id="fnref:12" class="footnote-ref"><a href="#fn:12" rel="footnote"><span class="hint--top hint--rounded" aria-label="[P25 25.RPKM FPKM and TPM Clearly Explained](https://www.bilibili.com/video/BV1yS4y1Z721/?p=25)">[12]</span></a></sup></h3><h4 id="计算步骤">计算步骤</h4><p>RPKM和FPKM计算方法一致，区别仅在于原始数据是否使用双端测序得来，其公式为：</p><p><span class="math display">\[ \begin{align} RPKM(x) &amp;= \frac{Reads\ per\ transcription}{\frac{totla\ reads}{10^6} \times \frac{transcription\ length}{1000}}\\ &amp;= \frac{10^9 \times C_x}{R \times L_x} \end{align} \]</span> 具体而言，可以分成三步：</p><ol type="1"><li>对read count求和，并除以1000000</li><li>将每一个read count除以上面的结果，得到RPM（read per million）</li><li>将RPM除以length再乘以1000，得到RPKM</li></ol><p>TPM类似，其计算公式为： <span class="math display">\[ TPM(X)=\frac{C_x \times r \times 10^6}{L_x \times T} \]</span> 具体步骤也可以分为三步：</p><ol type="1"><li>将每一个read count除以length，乘以1000，得到RPK</li><li>对RPK求和并除以1000000</li><li>再用RPK除以上面的结果，得到TPM</li></ol><h4 id="程序">程序</h4><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br></pre></td><td class="code"><pre><code class="hljs python"><span class="hljs-keyword">import</span> pandas <span class="hljs-keyword">as</span> pd<br><span class="hljs-keyword">import</span> matplotlib.pyplot <span class="hljs-keyword">as</span> plt<br><span class="hljs-keyword">import</span> numpy <span class="hljs-keyword">as</span> np<br><br>counts_df = pd.read_csv(<span class="hljs-string">&#x27;counts.txt&#x27;</span>, sep=<span class="hljs-string">&#x27;\t&#x27;</span>)<br><br><span class="hljs-built_in">print</span>(counts_df.columns)<br><br>express_num = <span class="hljs-built_in">len</span>(counts_df) - <span class="hljs-number">1</span><br><span class="hljs-built_in">print</span>(<span class="hljs-string">f&#x27;表达的基因数为:<span class="hljs-subst">&#123;express_num&#125;</span>&#x27;</span>)<br><br>count_10 = ((counts_df[<span class="hljs-string">&#x27;H1&#x27;</span>] &gt; <span class="hljs-number">10</span>) | (counts_df[<span class="hljs-string">&#x27;H2&#x27;</span>] &gt; <span class="hljs-number">10</span>)).<span class="hljs-built_in">sum</span>()<br><span class="hljs-built_in">print</span>(<span class="hljs-string">f&quot;有<span class="hljs-subst">&#123;count_10&#125;</span>个基因read count &gt; 10。&quot;</span>)<br><br><span class="hljs-comment"># FPKM</span><br>FPKM = pd.DataFrame()<br>FPKM[<span class="hljs-string">&#x27;Geneid&#x27;</span>] = counts_df[<span class="hljs-string">&#x27;Geneid&#x27;</span>]<br><br>FMP1 = counts_df[<span class="hljs-string">&#x27;H1&#x27;</span>] / (counts_df[<span class="hljs-string">&#x27;H1&#x27;</span>].<span class="hljs-built_in">sum</span>() / <span class="hljs-number">10</span> ** <span class="hljs-number">9</span>)<br>FMP2 = counts_df[<span class="hljs-string">&#x27;H2&#x27;</span>] / (counts_df[<span class="hljs-string">&#x27;H2&#x27;</span>].<span class="hljs-built_in">sum</span>() / <span class="hljs-number">10</span> ** <span class="hljs-number">9</span>)<br><br>FPKM[<span class="hljs-string">&#x27;H1&#x27;</span>] = (FMP1 / counts_df[<span class="hljs-string">&#x27;Length&#x27;</span>]) * <span class="hljs-number">10</span> ** <span class="hljs-number">3</span><br>FPKM[<span class="hljs-string">&#x27;H2&#x27;</span>] = (FMP2 / counts_df[<span class="hljs-string">&#x27;Length&#x27;</span>]) * <span class="hljs-number">10</span> ** <span class="hljs-number">3</span><br><br>FPKM.to_csv(<span class="hljs-string">&#x27;fpkm.csv&#x27;</span>, sep=<span class="hljs-string">&#x27;\t&#x27;</span>, index=<span class="hljs-literal">False</span>)<br><br><span class="hljs-comment"># TPM</span><br>TPM = pd.DataFrame()<br>TPM[<span class="hljs-string">&#x27;Geneid&#x27;</span>] = counts_df[<span class="hljs-string">&#x27;Geneid&#x27;</span>]<br><br>RPK1 = (counts_df[<span class="hljs-string">&#x27;H1&#x27;</span>] / counts_df[<span class="hljs-string">&#x27;Length&#x27;</span>]) * <span class="hljs-number">10</span> ** <span class="hljs-number">3</span><br>RPK2 = (counts_df[<span class="hljs-string">&#x27;H2&#x27;</span>] / counts_df[<span class="hljs-string">&#x27;Length&#x27;</span>]) * <span class="hljs-number">10</span> ** <span class="hljs-number">3</span><br><br>TPM[<span class="hljs-string">&#x27;H1&#x27;</span>] = RPK1 / (<span class="hljs-built_in">sum</span>(RPK1) / <span class="hljs-number">10</span> ** <span class="hljs-number">6</span>)<br>TPM[<span class="hljs-string">&#x27;H2&#x27;</span>] = RPK2 / (<span class="hljs-built_in">sum</span>(RPK2) / <span class="hljs-number">10</span> ** <span class="hljs-number">6</span>)<br><br>TPM.to_csv(<span class="hljs-string">&#x27;tpm.csv&#x27;</span>, sep=<span class="hljs-string">&#x27;\t&#x27;</span>, index=<span class="hljs-literal">False</span>)<br><br><span class="hljs-comment"># plot</span><br>data_1 = &#123;<span class="hljs-string">&#x27;H1&#x27;</span>: np.log2(FPKM[<span class="hljs-string">&#x27;H1&#x27;</span>] + <span class="hljs-number">1</span>),<br>          <span class="hljs-string">&#x27;H2&#x27;</span>: np.log2(FPKM[<span class="hljs-string">&#x27;H2&#x27;</span>] + <span class="hljs-number">1</span>)&#125;<br>FPKM_COPY = pd.DataFrame(data_1)<br>FPKM_COPY = FPKM_COPY.replace(<span class="hljs-number">0</span>, np.nan)<br><br>FPKM_COPY.plot(kind=<span class="hljs-string">&#x27;density&#x27;</span>)<br>plt.legend()<br>plt.xlabel(<span class="hljs-string">&#x27;log2(FPKM+1)&#x27;</span>)<br>plt.ylabel(<span class="hljs-string">&#x27;Density&#x27;</span>)<br>plt.title(<span class="hljs-string">&#x27;Density of FPKM&#x27;</span>)<br>plt.show()<br><br>data_2 = &#123;<span class="hljs-string">&#x27;H1&#x27;</span>: np.log2(TPM[<span class="hljs-string">&#x27;H1&#x27;</span>] + <span class="hljs-number">1</span>),<br>          <span class="hljs-string">&#x27;H2&#x27;</span>: np.log2(TPM[<span class="hljs-string">&#x27;H2&#x27;</span>] + <span class="hljs-number">1</span>)&#125;<br>TPM_COPY = pd.DataFrame(data_2)<br>TPM_COPY = TPM_COPY.replace(<span class="hljs-number">0</span>, np.nan)<br><br>TPM_COPY.plot(kind=<span class="hljs-string">&#x27;density&#x27;</span>)<br>plt.legend()<br>plt.xlabel(<span class="hljs-string">&#x27;log2(TPM+1)&#x27;</span>)<br>plt.ylabel(<span class="hljs-string">&#x27;Density&#x27;</span>)<br>plt.title(<span class="hljs-string">&#x27;Density of TPM&#x27;</span>)<br>plt.show()<br><br><span class="hljs-built_in">print</span>(<span class="hljs-string">&#x27;done&#x27;</span>)<br></code></pre></td></tr></table></figure><h2 id="参考">参考</h2><section class="footnotes"><div class="footnote-list"><ol><li><span id="fn:1" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://learn.microsoft.com/en-us/windows/wsl/install">https://learn.microsoft.com/en-us/windows/wsl/install</a> <a href="#fnref:1" rev="footnote" class="footnote-backref">↩︎</a></span></span></li><li><span id="fn:2" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://conda.io/projects/conda/en/stable/user-guide/install/index.html">https://conda.io/projects/conda/en/stable/user-guide/install/index.html</a> <a href="#fnref:2" rev="footnote" class="footnote-backref">↩︎</a></span></span></li><li><span id="fn:3" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://mirrors.tuna.tsinghua.edu.cn/help/anaconda/">https://mirrors.tuna.tsinghua.edu.cn/help/anaconda/</a> <a href="#fnref:3" rev="footnote" class="footnote-backref">↩︎</a></span></span></li><li><span id="fn:4" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://github.com/s-andrews/FastQC">https://github.com/s-andrews/FastQC</a> <a href="#fnref:4" rev="footnote" class="footnote-backref">↩︎</a></span></span></li><li><span id="fn:5" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://multiqc.info/">https://multiqc.info/</a> <a href="#fnref:5" rev="footnote" class="footnote-backref">↩︎</a></span></span></li><li><span id="fn:6" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://github.com/FelixKrueger/TrimGalore">https://github.com/FelixKrueger/TrimGalore</a> <a href="#fnref:6" rev="footnote" class="footnote-backref">↩︎</a></span></span></li><li><span id="fn:7" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://github.com/OpenGene/fastp">https://github.com/OpenGene/fastp</a> <a href="#fnref:7" rev="footnote" class="footnote-backref">↩︎</a></span></span></li><li><span id="fn:8" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://github.com/samtools/samtools/blob/develop/INSTALL">https://github.com/samtools/samtools/blob/develop/INSTALL</a> <a href="#fnref:8" rev="footnote" class="footnote-backref">↩︎</a></span></span></li><li><span id="fn:9" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/6b7a442d293f">samtools的安装和使用</a> <a href="#fnref:9" rev="footnote" class="footnote-backref">↩︎</a></span></span></li><li><span id="fn:10" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://zh.wikibooks.org/wiki/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/featureCounts">https://zh.wikibooks.org/wiki/生物信息学/featureCounts</a> <a href="#fnref:10" rev="footnote" class="footnote-backref">↩︎</a></span></span></li><li><span id="fn:11" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/545367874">基因组质量评估:(五)mapping法：2.samtools计算mapping rate</a> <a href="#fnref:11" rev="footnote" class="footnote-backref">↩︎</a></span></span></li><li><span id="fn:12" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://www.bilibili.com/video/BV1yS4y1Z721/?p=25">P25 25.RPKM FPKM and TPM Clearly Explained</a> <a href="#fnref:12" rev="footnote" class="footnote-backref">↩︎</a></span></span></li><li><span id="fn:13" class="footnote-text"><span><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/283198445">文件格式01：GTF与GFF文件</a> <a href="#fnref:13" rev="footnote" class="footnote-backref">↩︎</a></span></span></li></ol></div></section></div><hr><div><div class="post-metas my-3"><div class="post-meta mr-3 d-flex align-items-center"><i class="iconfont icon-category"></i> <span class="category-chains"><span class="category-chain"><a href="/categories/%E5%AD%A6%E4%B9%A0/" class="category-chain-item">学习</a> <span>></span> <a href="/categories/%E5%AD%A6%E4%B9%A0/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/" class="category-chain-item">生物信息学</a></span></span></div><div class="post-meta"><i class="iconfont icon-tags"></i> <a href="/tags/%E5%AD%A6%E4%B9%A0/" class="print-no-link">#学习</a> <a href="/tags/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6/" class="print-no-link">#生物信息学</a></div></div><div class="license-box my-3"><div class="license-title"><div>测序数据处理自用笔记</div><div>https://www.aye10032.com/2023/08/11/2023-08-11-trim_goal/</div></div><div class="license-meta"><div class="license-meta-item"><div>作者</div><div>Aye10032</div></div><div class="license-meta-item license-meta-date"><div>发布于</div><div>2023年8月11日</div></div><div class="license-meta-item license-meta-date"><div>更新于</div><div>2023年8月28日</div></div><div class="license-meta-item"><div>许可协议</div><div><a class="print-no-link" target="_blank" href="https://creativecommons.org/licenses/by/4.0/"><span class="hint--top hint--rounded" aria-label="BY - 署名"><i class="iconfont icon-by"></i></span></a></div></div></div><div class="license-icon iconfont"></div></div><div class="post-prevnext my-3"><article class="post-prev col-6"><a href="/2023/08/27/2023-08-27-docker/" title="Docker使用心得"><i class="iconfont icon-arrowleft"></i> <span class="hidden-mobile">Docker使用心得</span> <span class="visible-mobile">上一篇</span></a></article><article class="post-next col-6"><a href="/2023/05/24/2023-05-24-Bioinformatics/" title="生物信息学知识图谱"><span class="hidden-mobile">生物信息学知识图谱</span> <span class="visible-mobile">下一篇</span> <i class="iconfont icon-arrowright"></i></a></article></div></div><article id="comments" lazyload><div id="gitalk-container"></div><script type="text/javascript">Fluid.utils.loadComments("#gitalk-container",(function(){Fluid.utils.createCssLink("/css/gitalk.css"),Fluid.utils.createScript("https://lib.baomitu.com/gitalk/1.8.0/gitalk.min.js",(function(){var e=Object.assign({clientID:"eeda953af44e09f78441",clientSecret:"cb84da4697fdc74e72451e5b9618297b1667988b",repo:"Aye10032.github.io",owner:"Aye10032",admin:["Aye10032"],language:"zh-CN",labels:["Gitalk"],perPage:10,pagerDirection:"last",distractionFreeMode:!1,createIssueManually:!0,proxy:"https://cors-anywhere.azm.workers.dev/https://github.com/login/oauth/access_token"},{id:"5890787e1f7ae78283bde5619e6ea691"});new Gitalk(e).render("gitalk-container")}))}))</script><noscript>Please enable JavaScript to view the comments</noscript></article></article></div></div></div><div class="side-col d-none d-lg-block col-lg-2"><aside class="sidebar" style="margin-left:-1rem"><div id="toc"><p class="toc-header"><i class="iconfont icon-list"></i> <span>目录</span></p><div class="toc-body" id="toc-body"></div></div></aside></div></div></div><a id="scroll-top-button" aria-label="TOP" href="#" role="button"><i class="iconfont icon-arrowup" aria-hidden="true"></i></a><div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable modal-lg" role="document"><div class="modal-content"><div class="modal-header text-center"><h4 class="modal-title w-100 font-weight-bold">搜索</h4><button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button></div><div class="modal-body mx-3"><div class="md-form mb-5"><input type="text" id="local-search-input" class="form-control validate"> <label data-error="x" data-success="v" for="local-search-input">关键词</label></div><div class="list-group" id="local-search-result"></div></div></div></div></div></main><footer><div class="footer-inner"><div class="footer-content"><a href="https://hexo.io" target="_blank" rel="nofollow noopener"><span>Hexo</span></a> <a href="https://github.com/fluid-dev/hexo-theme-fluid" target="_blank" rel="nofollow noopener"><span>Fluid</span></a></div><div class="statistics"><span id="busuanzi_container_site_pv" style="display:none">总访问量 <span id="busuanzi_value_site_pv"></span> 次 </span><span id="busuanzi_container_site_uv" style="display:none">总访客数 <span id="busuanzi_value_site_uv"></span> 人</span></div><div class="beian"><span><a href="http://beian.miit.gov.cn/" target="_blank" rel="nofollow noopener">苏ICP备19040408号-2 </a></span><span><a href="http://www.beian.gov.cn/portal/registerSystemInfo?recordcode=32108102010462" rel="nofollow noopener" class="beian-police" target="_blank"><span style="visibility:hidden;width:0">|</span> <img src="/images/%E5%A4%87%E6%A1%88%E5%9B%BE%E6%A0%87.png" srcset="/images/loading.gif" lazyload alt="police-icon"> <span>苏公网安备 32108102010462号</span></a></span></div></div></footer><script src="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.js"></script><link rel="stylesheet" href="https://lib.baomitu.com/nprogress/0.2.0/nprogress.min.css"><script>NProgress.configure({showSpinner:!1,trickleSpeed:100}),NProgress.start(),window.addEventListener("load",(function(){NProgress.done()}))</script><script src="https://lib.baomitu.com/jquery/3.6.4/jquery.min.js"></script><script src="https://lib.baomitu.com/twitter-bootstrap/4.6.1/js/bootstrap.min.js"></script><script src="/js/events.js"></script><script src="/js/plugins.js"></script><script src="https://lib.baomitu.com/typed.js/2.0.12/typed.min.js"></script><script>!function(t,e){var i=Fluid.plugins.typing,n=e.getElementById("subtitle");n&&i&&i(n.getAttribute("data-typed-text"))}(window,document)</script><script src="/js/img-lazyload.js"></script><script>Fluid.utils.createScript("https://lib.baomitu.com/tocbot/4.20.1/tocbot.min.js",(function(){var t=jQuery("#toc");if(0!==t.length&&window.tocbot){var i=jQuery("#board-ctn").offset().top;window.tocbot.init(Object.assign({tocSelector:"#toc-body",contentSelector:".markdown-body",linkClass:"tocbot-link",activeLinkClass:"tocbot-active-link",listClass:"tocbot-list",isCollapsedClass:"tocbot-is-collapsed",collapsibleClass:"tocbot-is-collapsible",scrollSmooth:!0,includeTitleTags:!0,headingsOffset:-i},CONFIG.toc)),t.find(".toc-list-item").length>0&&t.css("visibility","visible"),Fluid.events.registerRefreshCallback((function(){if("tocbot"in window){tocbot.refresh();var t=jQuery("#toc");if(0===t.length||!tocbot)return;t.find(".toc-list-item").length>0&&t.css("visibility","visible")}}))}}))</script><script src="https://lib.baomitu.com/clipboard.js/2.0.11/clipboard.min.js"></script><script>Fluid.plugins.codeWidget()</script><script>Fluid.utils.createScript("https://lib.baomitu.com/anchor-js/4.3.1/anchor.min.js",(function(){window.anchors.options={placement:CONFIG.anchorjs.placement,visible:CONFIG.anchorjs.visible},CONFIG.anchorjs.icon&&(window.anchors.options.icon=CONFIG.anchorjs.icon);var n=(CONFIG.anchorjs.element||"h1,h2,h3,h4,h5,h6").split(","),o=[];for(var s of n)o.push(".markdown-body > "+s.trim());"left"===CONFIG.anchorjs.placement&&(window.anchors.options.class="anchorjs-link-left"),window.anchors.add(o.join(", ")),Fluid.events.registerRefreshCallback((function(){if("anchors"in window){anchors.removeAll();var n=(CONFIG.anchorjs.element||"h1,h2,h3,h4,h5,h6").split(","),o=[];for(var s of n)o.push(".markdown-body > "+s.trim());"left"===CONFIG.anchorjs.placement&&(anchors.options.class="anchorjs-link-left"),anchors.add(o.join(", "))}}))}))</script><script>Fluid.utils.createScript("https://lib.baomitu.com/fancybox/3.5.7/jquery.fancybox.min.js",(function(){Fluid.plugins.fancyBox()}))</script><script>Fluid.plugins.imageCaption()</script><script src="/js/local-search.js"></script><script defer src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><script src="/js/boot.js"></script><noscript><div class="noscript-warning">博客在允许 JavaScript 运行的环境下浏览效果更佳</div></noscript></body></html>